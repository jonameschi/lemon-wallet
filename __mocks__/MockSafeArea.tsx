import React, { FC, PropsWithChildren } from 'react';

import { Metrics, SafeAreaProvider } from 'react-native-safe-area-context';

const MOCK_INITIAL_METRICS: Metrics = {
  frame: {
    width: 320,
    height: 640,
    x: 0,
    y: 0,
  },
  insets: {
    left: 0,
    right: 0,
    bottom: 0,
    top: 0,
  },
};

type MockSafeAreaContextProps = {
  initialWindowMetrics?: Metrics | null;
};

const MockSafeAreaContext: FC<PropsWithChildren<MockSafeAreaContextProps>> = ({
  children,
  initialWindowMetrics,
}) => (
  <SafeAreaProvider
    initialMetrics={initialWindowMetrics || MOCK_INITIAL_METRICS}>
    {children}
  </SafeAreaProvider>
);

export { MockSafeAreaContext };
export type { MockSafeAreaContextProps };
