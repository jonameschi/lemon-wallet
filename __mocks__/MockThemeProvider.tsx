import { FC, PropsWithChildren } from 'react';
import { ThemeProvider } from 'styled-components/native';
import { theme } from 'theme/theme';

export const MockThemeProvider: FC<PropsWithChildren> = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);
