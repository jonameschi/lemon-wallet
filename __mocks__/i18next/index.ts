const useMock = (key: string): string => key;

export default {
  t: useMock,
};
