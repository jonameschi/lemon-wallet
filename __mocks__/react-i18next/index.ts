/* eslint-disable import/no-import-module-exports */
import { I18nextProvider, initReactI18next } from 'react-i18next';

const useMock = {
  t: (key: string, options: object | undefined): string =>
    options ? `${key} - ${JSON.stringify(options)}` : key,
};

module.exports = {
  useTranslation: () => useMock,
  I18nextProvider,
  initReactI18next,
};
