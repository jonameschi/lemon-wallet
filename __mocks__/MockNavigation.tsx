import React, { FunctionComponent, PropsWithChildren } from 'react';

import { NavigationContext } from '@react-navigation/native';
import { MockSafeAreaContext, MockSafeAreaContextProps } from './MockSafeArea';
import { MockThemeProvider } from './MockThemeProvider';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const mockNavigationProps: any = {
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: () => true,
  addListener: jest.fn(() => () => ({ remove: jest.fn() })),
};

type MockNavigationProps = PropsWithChildren<MockSafeAreaContextProps>;

const MockNavigation: FunctionComponent<MockNavigationProps> = ({
  children,
  initialWindowMetrics,
}) => (
  <MockSafeAreaContext initialWindowMetrics={initialWindowMetrics}>
    <NavigationContext.Provider value={mockNavigationProps}>
      <MockThemeProvider>{children}</MockThemeProvider>
    </NavigationContext.Provider>
  </MockSafeAreaContext>
);

export { MockNavigation, mockNavigationProps };
