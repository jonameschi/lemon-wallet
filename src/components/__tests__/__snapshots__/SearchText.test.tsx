import { ComponentProps } from 'react';
import { SearchText } from '@components';
import { fireEvent, render } from '@testing-library/react-native';
import { MockThemeProvider } from '../../../../__mocks__/MockThemeProvider';

const renderTree = (props: ComponentProps<typeof SearchText>) =>
  render(
    <MockThemeProvider>
      <SearchText {...props} />
    </MockThemeProvider>,
  );

describe('SearchText component', () => {
  const handleChange = jest.fn();

  it('should render a snapshot - default', () => {
    const { toJSON } = renderTree({
      onChangeText: handleChange,
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render a snapshot - with initial value and placeholder', () => {
    const { toJSON } = renderTree({
      onChangeText: handleChange,
      initialValue: 'Initial Value',
      placeholder: 'Placeholder',
    });
    expect(toJSON()).toMatchSnapshot();
  });

  it('should call to onChangeText when enter a text', () => {
    const { getByTestId } = renderTree({
      onChangeText: handleChange,
      initialValue: 'Initial Value',
      placeholder: 'Placeholder',
      testID: 'search-text-test',
    });

    fireEvent.press(getByTestId('search-text-test-clear-btn'));
    expect(handleChange).toHaveBeenCalledWith('');
  });
});
