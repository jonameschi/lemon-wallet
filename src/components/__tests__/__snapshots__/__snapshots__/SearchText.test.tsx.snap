// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchText component should render a snapshot - default 1`] = `
<View
  style={
    {
      "borderBottomRightRadius": 36,
      "borderColor": "#DFE2E4",
      "borderLeftWidth": 0,
      "borderTopRightRadius": 36,
      "borderWidth": 1,
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "marginBottom": 16,
      "marginTop": 16,
      "paddingBottom": 10,
      "paddingLeft": 20,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    alignItems="center"
    gap={8}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "flex-start",
      }
    }
  >
    <
      fill="#6C757D"
      height={24}
      width={24}
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      autoFocus={true}
      onChangeText={[Function]}
      placeholderTextColor="#6C757D"
      style={
        {
          "color": "#212529",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontSize": 16,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      testID="undefined-input"
    />
  </View>
</View>
`;

exports[`SearchText component should render a snapshot - with initial value and placeholder 1`] = `
<View
  style={
    {
      "borderBottomRightRadius": 36,
      "borderColor": "#DFE2E4",
      "borderLeftWidth": 0,
      "borderTopRightRadius": 36,
      "borderWidth": 1,
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "marginBottom": 16,
      "marginTop": 16,
      "paddingBottom": 10,
      "paddingLeft": 20,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    alignItems="center"
    gap={8}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "flex-start",
      }
    }
  >
    <
      fill="#6C757D"
      height={24}
      width={24}
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      autoFocus={true}
      onChangeText={[Function]}
      placeholder="Placeholder"
      placeholderTextColor="#6C757D"
      style={
        {
          "color": "#212529",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontSize": 16,
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      testID="undefined-input"
      value="Initial Value"
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="undefined-clear-btn"
    >
      <
        fill="#6C757D"
        height={24}
        width={24}
      />
    </View>
  </View>
</View>
`;
